src/main/java/solvd/laba/ermakovich/hu/command/DoctorCommandHandler.java

```diff
@@ -25,29 +26,30 @@ public class DoctorCommandHandler implements DoctorCommandService {
     private final BCryptPasswordEncoder bCryptPasswordEncoder;
     private final DoctorQueryService queryService;
     private final KafkaProducer kafkaProducer;
+    private final DoctorAggregateService aggregateService;
 
     @Override
     @SneakyThrows
-    public Mono<String> handle(CreateDoctorCommand command) {
+    public Mono<Void> handle(CreateDoctorCommand command) {
         command.getDoctor().setPassword(
-                hashPassword(command.getDoctor().getPassword()));
+                bCryptPasswordEncoder.encode(
+                        command.getDoctor().getPassword()
+                )
+        );
         return queryService.isExistByEmail(command.getDoctor().getEmail())
                 .flatMap(isExist -> {
                     if (Boolean.TRUE.equals(isExist)) {
                         throw new ResourceAlreadyExistsException("User with this email: " +
                                 command.getDoctor().getEmail() + " already exist");
                     } else {
-                        Event createDoctor = new CreateDoctor(command.getAggregateId(), command.getDoctor());
-                        eventService.when(createDoctor).subscribe();
-                        IntegrationEvent accountEvent = new CreateAccount(command.getDoctor().getExternalId());
+                        EventRoot createDoctor = new CreateDoctor(command.getAggregateId(), command.getDoctor());
+                        eventService.create(createDoctor);
+                        aggregateService.applyCreateOperation(createDoctor).subscribe();
+                        IntegrationEvent accountEvent = new CreateAccount(command.getDoctor().getExternalId(), command.getAggregateId());
                         kafkaProducer.send(accountEvent);
-                        return Mono.just(createDoctor.getAggregateId());
+                        return Mono.empty();
```

Comment:
it's not a good idea to return `Mono.empty()`

thus, you're saying that your software is actually `fail-safe` instead of raising exception or error(`fail-fast` approach)